### PR TITLE
[browser][file system] GetLogicalDrives implementation

### PR DESCRIPTION
**Pull request by kjpou1**
- Directory.GetLogicalDrives threw PNSE.  Follows existing code paths.
- Add common DriveInfoInternal.Browser that is common code path for other implementations
   - Environment.GetLogicalDrives
   - DriveInfo.Drives

Fix for FileSystemTest https://github.com/dotnet/runtime/blob/master/src/libraries/System.IO.FileSystem/tests/Directory/GetLogicalDrives.cs

### CODE REVIEW

File: src/libraries/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Browser.cs [hunk 1]
```diff
@@ -15,6 +15,6 @@ public sealed partial class DriveInfo
         public long TotalFreeSpace => 0;
         public long TotalSize => 0;
 
-        private static string[] GetMountPoints() => Environment.GetLogicalDrives();
+        private static string[] GetMountPoints() => DriveInfoInternal.GetLogicalDrives();
```
Comment (campersau) [inline src/libraries/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Browser.cs:18]: Is this change required since `Environment.GetLogicalDrives` returns the same value?

Comment (kjpou1) [reply src/libraries/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Browser.cs:18]: No not required but removes an extra call because `Environment.GetLogicalDrives()` calls `DriveInfoInternal.GetLogicalDrives();`

**Review by akoeplinger — APPROVED** [review]

File: src/libraries/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Browser.cs (hunk 1 above)
Comment (jkotas) [reply src/libraries/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Browser.cs:18]: Why does this extra call matter in this case? 

This is adding a second copy of DriveInfoInternal.Browser.cs that is much worse for Browser in particular than an extra call.

**Review by jkotas — CHANGES_REQUESTED** [review]

File: src/libraries/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Browser.cs (hunk 1 above)
Comment (akoeplinger) [reply src/libraries/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Browser.cs:18]: @jkotas not sure what you mean, this follows the existing pattern for `DriveInfoInternal.Unix.cs` that is already there.

Comment (jkotas) [reply src/libraries/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Browser.cs:18]: When having a choice, I think we should be optimizing Browser for smaller footprint, not for tiny bit better throughput of APIs that are slow by design and are very unlikely to be used in Browser.

I believe that the extra copy of `DriveInfoInternal.Browser.cs` is causing more harm here for browser than the extra call.

This does not matter much since the difference is small and all this code is going to be removed by the linker for browser anyway.

**Review by jkotas — APPROVED** [review]

**Comment by kjpou1** [comment]
Addressed in recent commits

File: src/libraries/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Browser.cs (hunk 1 above)
Comment (akoeplinger) [reply src/libraries/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Browser.cs]: Oh I see, you meant the extra copy of the implementation that is compiled into the library, I thought you meant some duplicate file :)

**Comment by akoeplinger** [comment]
The wasm test failure was due to https://github.com/dotnet/runtime/issues/39473
